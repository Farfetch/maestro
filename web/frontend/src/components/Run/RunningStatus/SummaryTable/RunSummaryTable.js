import { LineChartOutlined } from "@ant-design/icons";
import { Button, Table, Typography } from "antd";
import React from "react";

import { avg, calculateErrorRate } from "../../../../lib/utils";

const RunSummaryTable = ({
  metrics,
  isLoading,
  totals,
  setLabelToShowGraph,
  setActiveTabKey
}) => {
  const columns = [
    {
      title: "Label",
      dataIndex: "label",
      key: "label",
      width: 300,
      ellipsis: true,
      sorter: {
        compare: (recordA, recordB) => {
          if (recordA.url < recordB.label) {
            return -1;
          }
          if (recordA.label > recordB.label) {
            return 1;
          }
          return 0;
        }
      }
    },
    {
      title: "Latency (p50)",
      dataIndex: "latencyP50",
      key: "latencyP50",
      render: (text, record) => (
        <span key={`${record.url}-latencyP50`}>
          {`${record.latencyP50}`} <span style={{ fontSize: 11 }}>ms</span>
        </span>
      ),
      sorter: {
        compare: (recordA, recordB) => recordA.latencyP50 - recordB.latencyP50
      },
      width: 100
    },
    {
      title: "Latency (p99)",
      dataIndex: "latencyP99",
      key: "latencyP99",
      render: (text, record) => (
        <span key={`${record.url}-latencyP99`}>
          {`${record.latencyP99}`} <span style={{ fontSize: 11 }}>ms</span>
        </span>
      ),
      sorter: {
        compare: (recordA, recordB) => recordA.latencyP99 - recordB.latencyP99
      },
      width: 100
    },
    {
      title: "Total",
      dataIndex: "totalCount",
      key: "totalCount",
      render: (text, record) => (
        <span key={`${record.url}-totalCount`}>{`${record.totalCount}`}</span>
      ),
      sorter: {
        compare: (recordA, recordB) => recordA.totalCount - recordB.totalCount
      },
      width: 100
    },
    {
      title: "Success",
      dataIndex: "successCount",
      key: "successCount",
      render: (text, record) => (
        <span
          key={`${record.url}-successCount`}
        >{`${record.successCount}`}</span>
      ),
      sorter: {
        compare: (recordA, recordB) =>
          recordA.successCount - recordB.successCount
      },
      width: 100
    },
    {
      title: "Error count",
      dataIndex: "errorCount",
      key: "errorCount",
      render: (text, record) => (
        <span key={`${record.url}-error_count`}>{record.errorsCount}</span>
      ),
      sorter: {
        compare: (recordA, recordB) => recordA.errorsCount - recordB.errorsCount
      },
      width: 110
    },
    {
      title: "Error rate",
      dataIndex: "errorRate",
      key: "errorRate",
      defaultSortOrder: "descend",
      render: (text, record) => (
        <span key={`${record.url}-error_rate`}>
          {record.errorRate}
          <span style={{ fontSize: 11 }}>%</span>
        </span>
      ),
      sorter: {
        compare: (recordA, recordB) => recordA.errorRate - recordB.errorRate
      },
      width: 100
    },
    {
      title: "RPM",
      dataIndex: "rpm",
      key: "rpm",
      render: (text, record) => (
        <span key={`${record.url}-rpm`}>
          {record.rpm} <span style={{ fontSize: 11 }}>req/min</span>
        </span>
      ),
      sorter: {
        compare: (recordA, recordB) => recordA.rpm - recordB.rpm
      },
      width: 120
    },
    {
      key: "actions",
      width: 50,
      render: (text, record) => (
        <Button
          icon={<LineChartOutlined />}
          onClick={() => {
            setLabelToShowGraph(record.label);
            setActiveTabKey("endpoints");
          }}
        ></Button>
      )
    }
  ];

  const expandedRowRender = (record) => {
    const expendedRowColumns = [
      {
        title: "Response code",
        dataIndex: "responseCode",
        key: `${record.label}-responseCode-col`
      },
      {
        title: "Messages",
        key: `${record.label}-messages-col`,
        render: (text, { messages }) => (
          <span>
            {messages.map((message) => (
              <p key={message}>{message}</p>
            ))}
          </span>
        )
      },
      {
        title: "Success",
        dataIndex: "successCount",
        key: `${record.label}-successCount-col`
      },
      {
        title: "Total",
        dataIndex: "totalCount",
        key: `${record.label}-totalCount-col`
      },
      {
        title: "Errors",
        key: `${record.label}-errors-col`,
        render: (text, { successCount, totalCount }) => (
          <span>
            {calculateErrorRate(successCount, totalCount)}{" "}
            <span style={{ fontSize: 11 }}>%</span>
          </span>
        )
      }
    ];

    return (
      <span key={`${record.label}-extended-table`}>
        <Table
          columns={expendedRowColumns}
          dataSource={record.responses}
          pagination={false}
          rowKey="responseCode"
        />
        <Button
          icon={<LineChartOutlined />}
          onClick={() => {
            setLabelToShowGraph(record.label);
            setActiveTabKey("endpoints");
          }}
        ></Button>
      </span>
    );
  };

  return (
    <>
      <Table
        size="small"
        loading={isLoading}
        dataSource={metrics}
        columns={columns}
        pagination={false}
        expandable={{
          expandedRowRender
        }}
        summary={() => (
          <>
            <Table.Summary.Row>
              <Table.Summary.Cell />
              <Table.Summary.Cell>
                <Typography.Text strong>Total</Typography.Text>
              </Table.Summary.Cell>
              <Table.Summary.Cell>
                <Typography.Text>
                  <Typography.Text strong>
                    {avg(totals.totalLatencyP50, metrics.length).toFixed(2)}
                  </Typography.Text>
                  <span style={{ fontSize: 11 }}> ms</span>
                </Typography.Text>
              </Table.Summary.Cell>
              <Table.Summary.Cell>
                <Typography.Text>
                  <Typography.Text strong>
                    {avg(totals.totalLatencyP99, metrics.length).toFixed(2)}
                  </Typography.Text>
                  <span style={{ fontSize: 11 }}> ms</span>
                </Typography.Text>
              </Table.Summary.Cell>
              <Table.Summary.Cell>
                <Typography.Text strong>{totals.totalCount}</Typography.Text>
              </Table.Summary.Cell>
              <Table.Summary.Cell>
                <Typography.Text strong>
                  {totals.totalSuccessCount}
                </Typography.Text>
              </Table.Summary.Cell>
              <Table.Summary.Cell>
                <Typography.Text strong>
                  {totals.totalErrorsCount}
                </Typography.Text>
              </Table.Summary.Cell>
              <Table.Summary.Cell>
                <Typography.Text strong>
                  {parseFloat(totals.totalErrorsRate)}
                </Typography.Text>
                <span style={{ fontSize: 11 }}> %</span>
              </Table.Summary.Cell>
              <Table.Summary.Cell>
                <Typography.Text strong>
                  {avg(totals.totalRpm, metrics.length).toFixed(0)}
                </Typography.Text>
                <span style={{ fontSize: 11 }}> req/min</span>
              </Table.Summary.Cell>
            </Table.Summary.Row>
          </>
        )}
      />
    </>
  );
};

export default RunSummaryTable;
