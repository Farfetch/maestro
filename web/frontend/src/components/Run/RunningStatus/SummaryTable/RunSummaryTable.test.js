import { render } from "@testing-library/react";
import React from "react";

import RunSummaryTable from "./RunSummaryTable";

describe("RunSummaryTable", () => {
  const mockMetrics = [
    {
      label: "Label 1",
      latencyP50: 50,
      latencyP99: 60,
      totalCount: 100,
      successCount: 90,
      errorsCount: 10,
      errorRate: 10,
      rpm: 200,
      responses: [
        { responseCode: "200", messages: ["Success message 1"] },
        { responseCode: "400", messages: ["Error message 1"] }
      ]
    }
  ];

  const mockTotals = {
    totalLatencyP50: 55,
    totalLatencyP99: 65,
    totalCount: 200,
    totalSuccessCount: 180,
    totalErrorsCount: 20,
    totalErrorsRate: 10,
    totalRpm: 400
  };

  test("should render RunSummaryTable component", () => {
    const rendered = render(
      <RunSummaryTable
        metrics={mockMetrics}
        isLoading={false}
        totals={mockTotals}
        setLabelToShowGraph={() => {}}
        setActiveTabKey={() => {}}
      />
    );

    const component = rendered.container;
    expect(component.outerHTML).toMatchSnapshot();
  });
});
