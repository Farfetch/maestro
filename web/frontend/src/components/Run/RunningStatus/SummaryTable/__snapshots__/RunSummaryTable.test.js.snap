// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RunSummaryTable should render RunSummaryTable component 1`] = `"<div><div class=\\"ant-table-wrapper\\"><div class=\\"ant-spin-nested-loading\\"><div class=\\"ant-spin-container\\"><div class=\\"ant-table ant-table-small\\"><div class=\\"ant-table-container\\"><div class=\\"ant-table-content\\"><table style=\\"table-layout: fixed;\\"><colgroup><col class=\\"ant-table-expand-icon-col\\"><col style=\\"width: 300px;\\"><col style=\\"width: 100px;\\"><col style=\\"width: 100px;\\"><col style=\\"width: 100px;\\"><col style=\\"width: 100px;\\"><col style=\\"width: 110px;\\"><col style=\\"width: 100px;\\"><col style=\\"width: 120px;\\"><col style=\\"width: 50px;\\"></colgroup><thead class=\\"ant-table-thead\\"><tr><th class=\\"ant-table-cell ant-table-row-expand-icon-cell\\"></th><th class=\\"ant-table-cell ant-table-cell-ellipsis ant-table-column-has-sorters\\"><div class=\\"ant-table-column-sorters\\"><span class=\\"ant-table-column-title\\">Label</span><span class=\\"ant-table-column-sorter ant-table-column-sorter-full\\"><span class=\\"ant-table-column-sorter-inner\\"><span role=\\"img\\" aria-label=\\"caret-up\\" class=\\"anticon anticon-caret-up ant-table-column-sorter-up\\"><svg viewBox=\\"0 0 1024 1024\\" focusable=\\"false\\" data-icon=\\"caret-up\\" width=\\"1em\\" height=\\"1em\\" fill=\\"currentColor\\" aria-hidden=\\"true\\"><path d=\\"M858.9 689L530.5 308.2c-9.4-10.9-27.5-10.9-37 0L165.1 689c-12.2 14.2-1.2 35 18.5 35h656.8c19.7 0 30.7-20.8 18.5-35z\\"></path></svg></span><span role=\\"img\\" aria-label=\\"caret-down\\" class=\\"anticon anticon-caret-down ant-table-column-sorter-down\\"><svg viewBox=\\"0 0 1024 1024\\" focusable=\\"false\\" data-icon=\\"caret-down\\" width=\\"1em\\" height=\\"1em\\" fill=\\"currentColor\\" aria-hidden=\\"true\\"><path d=\\"M840.4 300H183.6c-19.7 0-30.7 20.8-18.5 35l328.4 380.8c9.4 10.9 27.5 10.9 37 0L858.9 335c12.2-14.2 1.2-35-18.5-35z\\"></path></svg></span></span></span></div></th><th class=\\"ant-table-cell ant-table-column-has-sorters\\"><div class=\\"ant-table-column-sorters\\"><span class=\\"ant-table-column-title\\">Latency (p50)</span><span class=\\"ant-table-column-sorter ant-table-column-sorter-full\\"><span class=\\"ant-table-column-sorter-inner\\"><span role=\\"img\\" aria-label=\\"caret-up\\" class=\\"anticon anticon-caret-up ant-table-column-sorter-up\\"><svg viewBox=\\"0 0 1024 1024\\" focusable=\\"false\\" data-icon=\\"caret-up\\" width=\\"1em\\" height=\\"1em\\" fill=\\"currentColor\\" aria-hidden=\\"true\\"><path d=\\"M858.9 689L530.5 308.2c-9.4-10.9-27.5-10.9-37 0L165.1 689c-12.2 14.2-1.2 35 18.5 35h656.8c19.7 0 30.7-20.8 18.5-35z\\"></path></svg></span><span role=\\"img\\" aria-label=\\"caret-down\\" class=\\"anticon anticon-caret-down ant-table-column-sorter-down\\"><svg viewBox=\\"0 0 1024 1024\\" focusable=\\"false\\" data-icon=\\"caret-down\\" width=\\"1em\\" height=\\"1em\\" fill=\\"currentColor\\" aria-hidden=\\"true\\"><path d=\\"M840.4 300H183.6c-19.7 0-30.7 20.8-18.5 35l328.4 380.8c9.4 10.9 27.5 10.9 37 0L858.9 335c12.2-14.2 1.2-35-18.5-35z\\"></path></svg></span></span></span></div></th><th class=\\"ant-table-cell ant-table-column-has-sorters\\"><div class=\\"ant-table-column-sorters\\"><span class=\\"ant-table-column-title\\">Latency (p99)</span><span class=\\"ant-table-column-sorter ant-table-column-sorter-full\\"><span class=\\"ant-table-column-sorter-inner\\"><span role=\\"img\\" aria-label=\\"caret-up\\" class=\\"anticon anticon-caret-up ant-table-column-sorter-up\\"><svg viewBox=\\"0 0 1024 1024\\" focusable=\\"false\\" data-icon=\\"caret-up\\" width=\\"1em\\" height=\\"1em\\" fill=\\"currentColor\\" aria-hidden=\\"true\\"><path d=\\"M858.9 689L530.5 308.2c-9.4-10.9-27.5-10.9-37 0L165.1 689c-12.2 14.2-1.2 35 18.5 35h656.8c19.7 0 30.7-20.8 18.5-35z\\"></path></svg></span><span role=\\"img\\" aria-label=\\"caret-down\\" class=\\"anticon anticon-caret-down ant-table-column-sorter-down\\"><svg viewBox=\\"0 0 1024 1024\\" focusable=\\"false\\" data-icon=\\"caret-down\\" width=\\"1em\\" height=\\"1em\\" fill=\\"currentColor\\" aria-hidden=\\"true\\"><path d=\\"M840.4 300H183.6c-19.7 0-30.7 20.8-18.5 35l328.4 380.8c9.4 10.9 27.5 10.9 37 0L858.9 335c12.2-14.2 1.2-35-18.5-35z\\"></path></svg></span></span></span></div></th><th class=\\"ant-table-cell ant-table-column-has-sorters\\"><div class=\\"ant-table-column-sorters\\"><span class=\\"ant-table-column-title\\">Total</span><span class=\\"ant-table-column-sorter ant-table-column-sorter-full\\"><span class=\\"ant-table-column-sorter-inner\\"><span role=\\"img\\" aria-label=\\"caret-up\\" class=\\"anticon anticon-caret-up ant-table-column-sorter-up\\"><svg viewBox=\\"0 0 1024 1024\\" focusable=\\"false\\" data-icon=\\"caret-up\\" width=\\"1em\\" height=\\"1em\\" fill=\\"currentColor\\" aria-hidden=\\"true\\"><path d=\\"M858.9 689L530.5 308.2c-9.4-10.9-27.5-10.9-37 0L165.1 689c-12.2 14.2-1.2 35 18.5 35h656.8c19.7 0 30.7-20.8 18.5-35z\\"></path></svg></span><span role=\\"img\\" aria-label=\\"caret-down\\" class=\\"anticon anticon-caret-down ant-table-column-sorter-down\\"><svg viewBox=\\"0 0 1024 1024\\" focusable=\\"false\\" data-icon=\\"caret-down\\" width=\\"1em\\" height=\\"1em\\" fill=\\"currentColor\\" aria-hidden=\\"true\\"><path d=\\"M840.4 300H183.6c-19.7 0-30.7 20.8-18.5 35l328.4 380.8c9.4 10.9 27.5 10.9 37 0L858.9 335c12.2-14.2 1.2-35-18.5-35z\\"></path></svg></span></span></span></div></th><th class=\\"ant-table-cell ant-table-column-has-sorters\\"><div class=\\"ant-table-column-sorters\\"><span class=\\"ant-table-column-title\\">Success</span><span class=\\"ant-table-column-sorter ant-table-column-sorter-full\\"><span class=\\"ant-table-column-sorter-inner\\"><span role=\\"img\\" aria-label=\\"caret-up\\" class=\\"anticon anticon-caret-up ant-table-column-sorter-up\\"><svg viewBox=\\"0 0 1024 1024\\" focusable=\\"false\\" data-icon=\\"caret-up\\" width=\\"1em\\" height=\\"1em\\" fill=\\"currentColor\\" aria-hidden=\\"true\\"><path d=\\"M858.9 689L530.5 308.2c-9.4-10.9-27.5-10.9-37 0L165.1 689c-12.2 14.2-1.2 35 18.5 35h656.8c19.7 0 30.7-20.8 18.5-35z\\"></path></svg></span><span role=\\"img\\" aria-label=\\"caret-down\\" class=\\"anticon anticon-caret-down ant-table-column-sorter-down\\"><svg viewBox=\\"0 0 1024 1024\\" focusable=\\"false\\" data-icon=\\"caret-down\\" width=\\"1em\\" height=\\"1em\\" fill=\\"currentColor\\" aria-hidden=\\"true\\"><path d=\\"M840.4 300H183.6c-19.7 0-30.7 20.8-18.5 35l328.4 380.8c9.4 10.9 27.5 10.9 37 0L858.9 335c12.2-14.2 1.2-35-18.5-35z\\"></path></svg></span></span></span></div></th><th class=\\"ant-table-cell ant-table-column-has-sorters\\"><div class=\\"ant-table-column-sorters\\"><span class=\\"ant-table-column-title\\">Error count</span><span class=\\"ant-table-column-sorter ant-table-column-sorter-full\\"><span class=\\"ant-table-column-sorter-inner\\"><span role=\\"img\\" aria-label=\\"caret-up\\" class=\\"anticon anticon-caret-up ant-table-column-sorter-up\\"><svg viewBox=\\"0 0 1024 1024\\" focusable=\\"false\\" data-icon=\\"caret-up\\" width=\\"1em\\" height=\\"1em\\" fill=\\"currentColor\\" aria-hidden=\\"true\\"><path d=\\"M858.9 689L530.5 308.2c-9.4-10.9-27.5-10.9-37 0L165.1 689c-12.2 14.2-1.2 35 18.5 35h656.8c19.7 0 30.7-20.8 18.5-35z\\"></path></svg></span><span role=\\"img\\" aria-label=\\"caret-down\\" class=\\"anticon anticon-caret-down ant-table-column-sorter-down\\"><svg viewBox=\\"0 0 1024 1024\\" focusable=\\"false\\" data-icon=\\"caret-down\\" width=\\"1em\\" height=\\"1em\\" fill=\\"currentColor\\" aria-hidden=\\"true\\"><path d=\\"M840.4 300H183.6c-19.7 0-30.7 20.8-18.5 35l328.4 380.8c9.4 10.9 27.5 10.9 37 0L858.9 335c12.2-14.2 1.2-35-18.5-35z\\"></path></svg></span></span></span></div></th><th aria-sort=\\"descending\\" class=\\"ant-table-cell ant-table-column-sort ant-table-column-has-sorters\\"><div class=\\"ant-table-column-sorters\\"><span class=\\"ant-table-column-title\\">Error rate</span><span class=\\"ant-table-column-sorter ant-table-column-sorter-full\\"><span class=\\"ant-table-column-sorter-inner\\"><span role=\\"img\\" aria-label=\\"caret-up\\" class=\\"anticon anticon-caret-up ant-table-column-sorter-up\\"><svg viewBox=\\"0 0 1024 1024\\" focusable=\\"false\\" data-icon=\\"caret-up\\" width=\\"1em\\" height=\\"1em\\" fill=\\"currentColor\\" aria-hidden=\\"true\\"><path d=\\"M858.9 689L530.5 308.2c-9.4-10.9-27.5-10.9-37 0L165.1 689c-12.2 14.2-1.2 35 18.5 35h656.8c19.7 0 30.7-20.8 18.5-35z\\"></path></svg></span><span role=\\"img\\" aria-label=\\"caret-down\\" class=\\"anticon anticon-caret-down ant-table-column-sorter-down active\\"><svg viewBox=\\"0 0 1024 1024\\" focusable=\\"false\\" data-icon=\\"caret-down\\" width=\\"1em\\" height=\\"1em\\" fill=\\"currentColor\\" aria-hidden=\\"true\\"><path d=\\"M840.4 300H183.6c-19.7 0-30.7 20.8-18.5 35l328.4 380.8c9.4 10.9 27.5 10.9 37 0L858.9 335c12.2-14.2 1.2-35-18.5-35z\\"></path></svg></span></span></span></div></th><th class=\\"ant-table-cell ant-table-column-has-sorters\\"><div class=\\"ant-table-column-sorters\\"><span class=\\"ant-table-column-title\\">RPM</span><span class=\\"ant-table-column-sorter ant-table-column-sorter-full\\"><span class=\\"ant-table-column-sorter-inner\\"><span role=\\"img\\" aria-label=\\"caret-up\\" class=\\"anticon anticon-caret-up ant-table-column-sorter-up\\"><svg viewBox=\\"0 0 1024 1024\\" focusable=\\"false\\" data-icon=\\"caret-up\\" width=\\"1em\\" height=\\"1em\\" fill=\\"currentColor\\" aria-hidden=\\"true\\"><path d=\\"M858.9 689L530.5 308.2c-9.4-10.9-27.5-10.9-37 0L165.1 689c-12.2 14.2-1.2 35 18.5 35h656.8c19.7 0 30.7-20.8 18.5-35z\\"></path></svg></span><span role=\\"img\\" aria-label=\\"caret-down\\" class=\\"anticon anticon-caret-down ant-table-column-sorter-down\\"><svg viewBox=\\"0 0 1024 1024\\" focusable=\\"false\\" data-icon=\\"caret-down\\" width=\\"1em\\" height=\\"1em\\" fill=\\"currentColor\\" aria-hidden=\\"true\\"><path d=\\"M840.4 300H183.6c-19.7 0-30.7 20.8-18.5 35l328.4 380.8c9.4 10.9 27.5 10.9 37 0L858.9 335c12.2-14.2 1.2-35-18.5-35z\\"></path></svg></span></span></span></div></th><th class=\\"ant-table-cell\\"></th></tr></thead><tbody class=\\"ant-table-tbody\\"><tr class=\\"ant-table-row ant-table-row-level-0\\"><td class=\\"ant-table-cell ant-table-row-expand-icon-cell\\"><button type=\\"button\\" class=\\"ant-table-row-expand-icon ant-table-row-expand-icon-collapsed\\" aria-label=\\"Expand row\\"></button></td><td title=\\"Label 1\\" class=\\"ant-table-cell ant-table-cell-ellipsis\\">Label 1</td><td class=\\"ant-table-cell\\"><span>50 <span style=\\"font-size: 11px;\\">ms</span></span></td><td class=\\"ant-table-cell\\"><span>60 <span style=\\"font-size: 11px;\\">ms</span></span></td><td class=\\"ant-table-cell\\"><span>100</span></td><td class=\\"ant-table-cell\\"><span>90</span></td><td class=\\"ant-table-cell\\"><span>10</span></td><td class=\\"ant-table-cell ant-table-column-sort\\"><span>10<span style=\\"font-size: 11px;\\">%</span></span></td><td class=\\"ant-table-cell\\"><span>200 <span style=\\"font-size: 11px;\\">req/min</span></span></td><td class=\\"ant-table-cell\\"><button type=\\"button\\" class=\\"ant-btn ant-btn-default ant-btn-icon-only\\"><span role=\\"img\\" aria-label=\\"line-chart\\" class=\\"anticon anticon-line-chart\\"><svg viewBox=\\"64 64 896 896\\" focusable=\\"false\\" data-icon=\\"line-chart\\" width=\\"1em\\" height=\\"1em\\" fill=\\"currentColor\\" aria-hidden=\\"true\\"><path d=\\"M888 792H200V168c0-4.4-3.6-8-8-8h-56c-4.4 0-8 3.6-8 8v688c0 4.4 3.6 8 8 8h752c4.4 0 8-3.6 8-8v-56c0-4.4-3.6-8-8-8zM305.8 637.7c3.1 3.1 8.1 3.1 11.3 0l138.3-137.6L583 628.5c3.1 3.1 8.2 3.1 11.3 0l275.4-275.3c3.1-3.1 3.1-8.2 0-11.3l-39.6-39.6a8.03 8.03 0 00-11.3 0l-230 229.9L461.4 404a8.03 8.03 0 00-11.3 0L266.3 586.7a8.03 8.03 0 000 11.3l39.5 39.7z\\"></path></svg></span></button></td></tr></tbody><tfoot class=\\"ant-table-summary\\"><tr><td class=\\"ant-table-cell\\"></td><td class=\\"ant-table-cell\\"><span class=\\"ant-typography\\"><strong>Total</strong></span></td><td class=\\"ant-table-cell\\"><span class=\\"ant-typography\\"><span class=\\"ant-typography\\"><strong>55.00</strong></span><span style=\\"font-size: 11px;\\"> ms</span></span></td><td class=\\"ant-table-cell\\"><span class=\\"ant-typography\\"><span class=\\"ant-typography\\"><strong>65.00</strong></span><span style=\\"font-size: 11px;\\"> ms</span></span></td><td class=\\"ant-table-cell\\"><span class=\\"ant-typography\\"><strong>200</strong></span></td><td class=\\"ant-table-cell\\"><span class=\\"ant-typography\\"><strong>180</strong></span></td><td class=\\"ant-table-cell\\"><span class=\\"ant-typography\\"><strong>20</strong></span></td><td class=\\"ant-table-cell\\"><span class=\\"ant-typography\\"><strong>10</strong></span><span style=\\"font-size: 11px;\\"> %</span></td><td class=\\"ant-table-cell\\"><span class=\\"ant-typography\\"><strong>400</strong></span><span style=\\"font-size: 11px;\\"> req/min</span></td></tr></tfoot></table></div></div></div></div></div></div></div>"`;
